test/src/instabug_modular_manager_test.dart

```diff
@@ -0,0 +1,202 @@
+import 'package:flutter/widgets.dart';
+import 'package:flutter_modular/flutter_modular.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:instabug_flutter_modular/instabug_flutter_modular.dart';
+import 'package:instabug_flutter_modular/src/instabug_modular_manager.dart';
+//ignore_for_file:depend_on_referenced_packages
+import 'package:modular_interfaces/src/route/modular_arguments.dart';
+
+import 'package:instabug_flutter/instabug_flutter.dart';
+
+import 'package:mockito/annotations.dart';
+
+import 'instabug_modular_manager_test.mocks.dart';
+
+@GenerateNiceMocks([
+  MockSpec<Module>(),
+  MockSpec<InstabugModule>(),
+  MockSpec<BuildContext>(),
+  MockSpec<ModularArguments>(),
+  MockSpec<Widget>(),
+  MockSpec<CustomTransition>(),
+])
+void main() {
+  late MockInstabugModule mockInstabugModule;
+  late MockModule mockModule;
+  late MockWidget mockWidget;
+  late MockBuildContext mockContext;
+  late MockModularArguments mockArgs;
+
+  setUp(() {
+    mockInstabugModule = MockInstabugModule();
+    mockModule = MockModule();
+    mockWidget = MockWidget();
+    mockContext = MockBuildContext();
+    mockArgs = MockModularArguments();
+  });
+
+  test('[wrapRoutes] with simple list', () {
+    // Arrange
+    final routes = [
+      ChildRoute('/home', child: (_, __) => mockWidget),
+      ModuleRoute('/profile', module: mockModule),
+    ];
+
+    // Act
+    final wrappedRoutes = InstabugModularManager.instance.wrapRoutes(routes);
+
+    // Assert
+    expect(wrappedRoutes.length, 2);
+    expect(wrappedRoutes[0].name, '/home');
+    expect(wrappedRoutes[1].name, '/profile');
+  });
+
+  test('[wrapRoutes] with nested route', () {
+    // Arrange
+    final routes = [
+      ChildRoute(
+        '/users',
+        child: (_, __) => mockWidget,
+        children: [
+          ModuleRoute('/list', module: mockModule),
+          ModuleRoute('/details', module: mockModule),
+        ],
+      ),
+    ];
+
+    // Act
+    final wrappedRoutes = InstabugModularManager.instance.wrapRoutes(routes);
+
+    // Assert
+    expect(wrappedRoutes.length, 1);
+    expect(wrappedRoutes[0].name, '/users');
+    expect(wrappedRoutes[0].children.length, 2);
+    expect(wrappedRoutes[0].children[0].name, '/list');
+    expect(wrappedRoutes[0].children[1].name, '/details');
+  });
+
+  test('[wrapRoute] with [ModuleRoute] should wrap with [InstabugModule]', () {
+    // Arrange
+    final route = ModuleRoute('/home', module: mockInstabugModule);
+
+    // Act
+    final wrappedRoute =
+        InstabugModularManager.instance.wrapRoute(route) as dynamic;
```

Comment:
Why do we need `as dynamic` here 🤔

Reply:
Flutter Modular has some issues in its object structure. For example, only the child route has a child property, and the return of the wrapRoute() method is the base class, which is ModularRoute. As a workaround, we cast it to dynamic to access the desired props without compile-time errors, such as 'this prop is not defined in ModularRoute'.